sql/mysql/sqlspec.go

```diff
@@ -0,0 +1,222 @@
+package mysql
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"ariga.io/atlas/schema/schemaspec"
+	"ariga.io/atlas/sql/internal/specutil"
+	"ariga.io/atlas/sql/schema"
+	"ariga.io/atlas/sql/sqlspec"
+)
+
+// UnmarshalSpec unmarshals an Atlas DDL document using an unmarshaler into v.
+func UnmarshalSpec(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{}) error {
+	var doc struct {
+		Tables  []*sqlspec.Table  `spec:"table"`
+		Schemas []*sqlspec.Schema `spec:"schema"`
+	}
+	if err := unmarshaler.UnmarshalSpec(data, &doc); err != nil {
+		return err
+	}
+	switch v := v.(type) {
```

Comment:
Is there any reason for using a type switch and not an if statement? Do you plan to add more cases in the future?